app/controllers/concerns/order_completion.rb

```diff
@@ -87,7 +87,6 @@ def processing_failed(error = RuntimeError.new(order_processing_error))
       payload.add_metadata :order, @order
     end
     flash[:error] = order_processing_error if flash.blank?
-    Checkout::PostCheckoutActions.new(@order).failure
```

Comment:
I think that this line is quite important in other cases. For example, if something goes out of stock or the order cycle closes then we need to reset some data. I can't remember the exact circumstances but we had bugs without it. This is actually the only line calling the `failure` method. So maybe you need two methods:

* `processing_failed` and
* `payment_failed`

`payment_failed` would be the Bugsnag notification and flash error while `processing_failed` also calls the `failure` method.

Reply:
It seems like we want to treat "processing failures" consistently with `processing_failed`, but not all are to be treated with `PostCheckoutActions#failure`. If it is important, I would have expected that to be apparent in the controller or concern specs. So maybe they need updating too!

Maybe that method should be renamed more specifically about what failures it's for (I've not looked into what it's for yet!).
And so as Maikel suggests, we only call it if it's not a payment error.

Reply:
Ok, I extracted that call to `processing_failed`